tests/testthat/test-derive_vars_dy.R

```diff
@@ -0,0 +1,216 @@
+#----test 1----
+# nolint start
+
+  test_that("Test1::Single --DT input when ref date is --DTM", {
+    datain <- tibble::tribble(
+      ~STUDYID, ~USUBJID, ~TRTSDTM, ~ASTDT,
+      "TEST01", "PAT01", "2014-01-17T23:59:59","2014-01-18" ,
+
+    ) %>%
+      mutate(TRTSDTM=lubridate::as_datetime(TRTSDTM),
+             ASTDT=lubridate::ymd(ASTDT))
+
+    expected_output <- tibble::tribble(
+      ~STUDYID, ~USUBJID, ~TRTSDTM, ~ASTDT,  ~ASTDY,
+      "TEST01", "PAT01", "2014-01-17T23:59:59","2014-01-18", 2
+
+    ) %>%
+      mutate(TRTSDTM=lubridate::as_datetime(TRTSDTM),
+             ASTDT=lubridate::ymd(ASTDT))
+
+      actual_output <- derive_vars_dy(datain,
+                                            reference_date = TRTSDTM,
+                                            source_vars = vars(ASTDT))
+
+            expect_dfs_equal(
+              expected_output,
+              actual_output,
+              keys = c("STUDYID", "USUBJID")
+            )
+          })
+
+#----test 2----
+test_that("Test2::Multiple --DT input when ref date is --DTM", {
+  datain <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~TRTSDTM, ~ASTDT, ~AENDT,
+    "TEST01", "PAT01", "2014-01-17T23:59:59", "2014-01-18", "2014-01-20"
+
+  ) %>%
+    mutate(TRTSDTM=lubridate::as_datetime(TRTSDTM),
+           ASTDT=lubridate::ymd(ASTDT),
+           AENDT=lubridate::ymd(AENDT))
+
+  expected_output <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~TRTSDTM, ~ASTDT, ~AENDT,  ~TRTSDY, ~ASTDY, ~AENDY,
+    "TEST01", "PAT01", "2014-01-17T23:59:59", "2014-01-18", "2014-01-20", 1, 2, 4
+
+  ) %>%
+    mutate(TRTSDTM=lubridate::as_datetime(TRTSDTM),
+           ASTDT=lubridate::ymd(ASTDT),
+           AENDT=lubridate::ymd(AENDT))
+
+  actual_output <- derive_vars_dy(datain,
+                                  reference_date = TRTSDTM,
+                                  source_vars = vars(TRTSDTM, ASTDT, AENDT))
```

Comment:
Judging from the test description this should be `TRTSDT`, not `TRTSDTM` for `source_vars`.

Reply:
The idea here is that the reference date - trst st date is a --DTM, but the other dates are --DT, But the user might still want the DY of reference date - I can change it and remove the TRTSDTM and keep the ASTDT and AENDT?